leaks.go

```diff
@@ -59,9 +59,6 @@ func Find(options ...Option) error {
 	if opts.cleanup != nil {
 		return errors.New("Cleanup can only be passed to VerifyNone or VerifyTestMain")
 	}
-	if opts.runOnFailure {
```

Comment:
This will also block its use in `VerifyTestMain`
I adjusted the doc to mention this has no effect on `VerifyNone`

keeping/adding a check in the code would require more effort/changes.